apps/backend/src/libs/modules/plugins/auth-plugin.ts

```diff
@@ -0,0 +1,42 @@
+import { type FastifyReply, type FastifyRequest } from "fastify";
+
+import { AuthErrorMessage } from "~/libs/enums/enums.js";
+import { HTTPCode } from "~/libs/modules/http/http.js";
+
+import { token } from "../token/token.js";
+import { OPEN_ROUTES } from "./libs/constants/constants.js";
+
+const authPlugin = async (
```

Comment:
Use [Fastify Plugins](https://fastify.dev/docs/latest/Reference/Plugins/) to create a plugin, this is a separate lib. You need to register them then, not to use preHandlers